ios/engine/KMEI/KeymanEngine/Classes/Keyboard/InputViewController.swift

```diff
@@ -333,6 +333,60 @@ open class InputViewController: UIInputViewController, KeymanWebDelegate {
       }
     }
   }
+  
+  func deleteSelection() -> Bool {
+    if let selected = textDocumentProxy.selectedText, selected.count > 0 {
+      /*
+        Since we're doing some funky text manipulation, it's best to add
+        a "canary" check in case something does go awry with it in
+        the future.
+      */
+      let beforeManipulation = textDocumentProxy.documentContextBeforeInput ?? ""
+
+      /*
+        Apple has some special nuances to its text deletion that our internal
+        Web engine does not emulate.
+       
+        .deleteBackward() behaviors:
+        - If there is selected text immediately following a space (U+0020),
+          it will delete that space IN ADDITION to the selected text.
+        - If there is selected text that starts mid-word, it will NOT delete
+          the preceding character.
+       
+        Compare to Web:  Web states an exact number of characters to delete
+        before the start of the currently-selected range... and it does this
+        completely unaware of the nuances listed above for .deleteBackward().
+        Keyman keyboard rules are likewise unaware of Apple's nuances.
+       
+        In order to maintain proper synchronization between app context and
+        internal Web-engine context, we need to force selected-text deletion
+        to NEVER delete preceding spaces.  Any attempts to adjust and include
+        the aforementioned nuance will need considerable design work to "get
+        right" due to the risk for adverse affects with Keyman keyboard rules.
```

Comment:
```suggestion
```

I don't think this comment is needed. Just the technical background.

Reply:
... this _is_ technical background, just from a different starting point.  As the primary maintainer of the Web/mobile-app interface, _this_ was the reason I made the change, after all - I wasn't even aware of the decision from #9073 when I made this.  It just so happens that both reasons align.